dartagnan/src/main/java/com/dat3m/dartagnan/program/analysis/alias/FieldSensitiveAndersen.java

```diff
@@ -209,12 +209,19 @@ private static final class Location {
 
         @Override
         public int hashCode() {
-            return base.hashCode() + Objects.hashCode(offset);
+            return 1201 * base.hashCode() + offset; // High factor to prevent overlapping hashcodes
         }
 
         @Override
         public boolean equals(Object o) {
-            return this == o || o instanceof Location && base.equals(((Location)o).base) && offset==((Location)o).offset;
+            if (this == o) {
+                return true;
+            } else if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            Location other = (Location) o;
+            // Can we check reference-equality on the MemoryObject?
```

Comment:
Is this a kind of "TODO question"?

Reply:
Almost, but it is not really important at all. I was just wondering if we ever create duplicated `MemoryObjects`, cause if not, then we could get away with reference equality checks. But this is no performance bottleneck, so it doesn't matter.